test/functional/tests/lazy_writes/test_lazy_writes_clean.py

```diff
@@ -0,0 +1,369 @@
+        cache = casadm.start_cache(cache_part, cache_mode)
+
+    with TestRun.step("Add core to cache."):
+        core = cache.add_core(core_part)
+
+    with TestRun.step("Disable cleaning and sequential cutoff policies."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Read IO stats before test"):
+        core_disk_writes_initial = check_device_write_stats(core_part)
+        exp_obj_writes_initial = check_device_write_stats(core)
+
+    with TestRun.step("Write data to the exported object."):
+        test_file_main = create_random_test_file("/tmp/test_file_main", Size(64, Unit.MebiByte))
+        dd = Dd().output(core.system_path) \
+            .input(test_file_main.full_path) \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+        test_file_md5sum_main = test_file_main.md5sum()
+
+    with TestRun.step("Check IO stats generated by previous step."):
+        core_disk_writes_current = (
+                check_device_write_stats(core_part) - core_disk_writes_initial
+        )
+        exp_obj_writes_current = (
+                check_device_write_stats(core) - exp_obj_writes_initial
+        )
+
+        if is_greater_than_zero(core_disk_writes_current):
+            TestRun.LOGGER.error("Writes should occur only on the exported object.")
+        if not is_equal(exp_obj_writes_current, test_file_main.size.value):
+            TestRun.LOGGER.error("Not all writes reached the exported object.")
+
+    with TestRun.step("Read data from the exported object."):
+        test_file_1 = File.create_file("/tmp/test_file_1")
+        dd = Dd().output(test_file_1.full_path) \
+            .input(core.system_path) \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+        test_file_1.refresh_item()
+        sync()
+
+    with TestRun.step("Stop cache."):
+        cache.stop()
+
+    with TestRun.step("Compare md5 sum of test files."):
+        if not is_equal(test_file_md5sum_main, test_file_1.md5sum()):
+            TestRun.LOGGER.error("Md5 sums should be equal.")
+
+    with TestRun.step("Read data from the core device."):
+        test_file_2 = File.create_file("/tmp/test_file_2")
+        dd = Dd().output(test_file_2.full_path) \
+            .input(core_part.system_path) \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+        test_file_2.refresh_item()
+        sync()
+
+    with TestRun.step("Compare md5 sum of test files."):
+        if not is_equal(test_file_md5sum_main, test_file_2.md5sum()):
+            TestRun.LOGGER.error("Md5 sums should be equal.")
+
+    with TestRun.step("Check IO stats generated for the core device since last check."):
```

Comment:
This is misleading - I don't see anywhere reassigning core_disk_writes_initial, so you are checking increase in relation to initial value, and not since the last check.

Reply:
Sorry - before I focused on step description matching actual action and not on which is correct. It looks like you want to check here if there were writes during cache stop. In that case, the previous description was ok, but the code was not doing that.